--- ../src-base/minecraft/net/minecraft/block/state/BlockPistonStructureHelper.java
+++ ../src-work/minecraft/net/minecraft/block/state/BlockPistonStructureHelper.java
@@ -10,7 +10,13 @@
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
+import carpet.CarpetSettings;
 
+
+import carpet.CarpetSettings;
+
+ import carpet.CarpetSettings;
+
 public class BlockPistonStructureHelper
 {
     private final World field_177261_a;
@@ -100,7 +106,7 @@
         {
             int i = 1;
 
-            if (i + this.field_177258_e.size() > 12)
+            if (i + this.field_177258_e.size() > CarpetSettings.pushLimit) //CM
             {
                 return false;
             }
@@ -119,7 +125,7 @@
 
                     ++i;
 
-                    if (i + this.field_177258_e.size() > 12)
+                    if (i + this.field_177258_e.size() > CarpetSettings.pushLimit) //CM
                     {
                         return false;
                     }
@@ -175,7 +181,7 @@
                         return true;
                     }
 
-                    if (this.field_177258_e.size() >= 12)
+                    if (this.field_177258_e.size() >= CarpetSettings.pushLimit) //CM
                     {
                         return false;
                     }
