--- ../src-base/minecraft/net/minecraft/world/NextTickListEntry.java
+++ ../src-work/minecraft/net/minecraft/world/NextTickListEntry.java
@@ -2,7 +2,11 @@
 
 import net.minecraft.block.Block;
 import net.minecraft.util.math.BlockPos;
+import carpet.CarpetSettings;
 
+
+import carpet.CarpetSettings;
+
 public class NextTickListEntry implements Comparable<NextTickListEntry>
 {
     private static long field_77177_f;
@@ -50,6 +54,8 @@
 
     public int compareTo(NextTickListEntry p_compareTo_1_)
     {
+        if (CarpetSettings.doubleTileTickSchedulingFix && equals(p_compareTo_1_))
+            return 0;
         if (this.field_77180_e < p_compareTo_1_.field_77180_e)
         {
             return -1;
