--- ../src-base/minecraft/net/minecraft/command/ServerCommandManager.java
+++ ../src-work/minecraft/net/minecraft/command/ServerCommandManager.java
@@ -32,7 +32,40 @@
 import net.minecraft.util.text.ITextComponent;
 import net.minecraft.util.text.TextComponentTranslation;
 import net.minecraft.util.text.TextFormatting;
+import carpet.CarpetServer;
+import carpet.commands.CommandAutosave;
+import carpet.commands.CommandBlockInfo;
+import carpet.commands.CommandCarpet;
+import carpet.commands.CommandCounter;
+import carpet.commands.CommandDistance;
+import carpet.commands.CommandEntityInfo;
+import carpet.commands.CommandFillBiome;
+import carpet.commands.CommandGMC;
+import carpet.commands.CommandGMS;
+import carpet.commands.CommandGrow;
+import carpet.commands.CommandLight;
+import carpet.commands.CommandLog;
+import carpet.commands.CommandPerimeter;
+import carpet.commands.CommandPing;
+import carpet.commands.CommandPlayer;
+import carpet.commands.CommandRNG;
+import carpet.commands.CommandRemoveEntity;
+import carpet.commands.CommandSpawn;
+import carpet.commands.CommandStructure;
+import carpet.commands.CommandTNT;
+import carpet.commands.CommandTick;
+import carpet.commands.CommandTickHealth;
+import carpet.commands.CommandTickingArea;
+import carpet.commands.CommandUnload;
+import carpet.commands.CommandUnload13;
+import carpet.commands.CommandWaypoint;
+import narcolepticfrog.rsmm.MeterCommand;
 
+
+import carpet.commands.*;
+import narcolepticfrog.rsmm.MeterCommand;
+import carpet.CarpetServer;
+ 
 public class ServerCommandManager extends CommandHandler implements ICommandListener
 {
     private final MinecraftServer field_184880_a;
@@ -89,7 +122,40 @@
         this.func_71560_a(new CommandLocate());
         this.func_71560_a(new CommandReload());
         this.func_71560_a(new CommandFunction());
+        
+        /* carpet mod */
+        this.func_71560_a(new CommandEntityInfo());
+        this.func_71560_a(new CommandSpawn());
+        this.func_71560_a(new CommandTick());
+        this.func_71560_a(new CommandTickHealth());
+        this.func_71560_a(new CommandDistance());
+        this.func_71560_a(new CommandBlockInfo());
+        this.func_71560_a(new CommandCarpet());
+        this.func_71560_a(new CommandCounter());
+        this.func_71560_a(new CommandUnload());
+        this.func_71560_a(new CommandUnload13());
+        this.func_71560_a(new CommandGMC());
+        this.func_71560_a(new CommandGMS());
+        this.func_71560_a(new CommandGrow());
+        this.func_71560_a(new CommandLog());
+        this.func_71560_a(new CommandPerimeter());
+        this.func_71560_a(new CommandRNG());
+        this.func_71560_a(new CommandTickingArea());
+        this.func_71560_a(new CommandStructure());
+        this.func_71560_a(new CommandFillBiome());
+        this.func_71560_a(new CommandPing());
+        this.func_71560_a(new CommandRemoveEntity());
 
+        this.func_71560_a(new CommandPlayer());
+        this.func_71560_a(new CommandAutosave());
+        this.func_71560_a(new CommandWaypoint());
+        this.func_71560_a(new CommandTNT());
+        this.func_71560_a(new CommandLight());
+        /* end */
+        // ----- RSMM Start ----- //
+        this.func_71560_a(new MeterCommand(CarpetServer.rsmmServer));
+        // ----- RSMM End ----- //
+
         if (p_i46985_1_.func_71262_S())
         {
             this.func_71560_a(new CommandOp());
